test/unit/middleware/redirects.spec.js

```diff
@@ -58,6 +58,38 @@ describe('redirect middleware', function() {
       .end(done);
   });
 
+  it('recognizes glob as synonymous with source', function(done) {
+    var app = connect()
+      .use(setup)
+      .use(redirect({redirects: [{
+        glob: '/source',
+        destination: '/redirect',
+        type: 301
+      }]}));
+
+    request(app)
+      .get('/source')
+      .expect(301)
+      .expect('location', '/redirect')
+      .end(done);
+  });
+
+  it('redirects to a configured regexp path', function(done) {
```

Comment:
I want a few more test cases here, but first: how should we deal with the fact that we are 100% going to have test cases that fail if an optional dependency isn't installed? Is it OK to make re2 a devDependency in addition to an optionalDependency in package.json?

Reply:
I'm not sure what the best practice here is going to be. I think if we are only going to be able to support unix-type environments with RE2, that's just what it's going to have to be. Keep it an optional dependency but don't write a test that would break if it's not installed.

@mbleigh do you have opinions here?

Reply:
It's fine to add an `optionalDependency` as a `devDependency` and test against it. If you want a way for the test suite to be able to test both states, you could consider an env variable specifically to disable RE2 but I'm probably fine with saying RE2 is the "correct" path and we're only supporting non-RE2 for the sake of install convenience.

It's not that RE2 can't be installed on non-unix-type environments, it's just that native dependencies *tend* to break more often on non-unix-type environments.